src/main/java/org/purescript/psi/ModuleReference.kt

```diff
@@ -0,0 +1,24 @@
+package org.purescript.psi
+
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiReferenceBase
+import com.intellij.psi.search.FilenameIndex
+import com.intellij.psi.search.GlobalSearchScope
+import org.purescript.file.PSFile
+
+class ModuleReference(element: PSImportDeclarationImpl) : PsiReferenceBase<PSImportDeclarationImpl>(
+    element,
+    TextRange.allOf(element.text.trim()),
+    false
+) {
+    override fun resolve(): PSModule? {
+        val psFile = FilenameIndex.getFilesByName(
+            myElement.project,
+            myElement.importName.split(".").last() + ".purs",
```

Comment:
This looks like you're assuming that the path for modules have to match their name? (Unless I'm mistaken). That's not the case though, the only file names the compiler cares about are FFI files being matched to their corresponding purs modules.